network-api/networkapi/wagtailpages/pagemodels/buyersguide/products.py

```diff
@@ -293,26 +293,71 @@ class ProductPageEvaluation(models.Model):
 
     @property
     def total_votes(self):
+        """Total number of votes for this product evaluation.
+
+        To populate the prefetched `_total_votes` annotation, call
+
+        ```
+        ProductPageEvaluation.objects.with_total_votes()
+        ```
+
+        Returns:
+            int: Total number of votes for this product evaluation.
+        """
         try:
             return self._total_votes
         except AttributeError:
             return self.votes.count()
 
     @property
     def total_creepiness(self):
+        """Aggregate of vote values for this product evaluation.
+
+        To use this property, make sure to first populate the prefetched `_total_creepiness`
+        annotation, i.e.:
+
+        ```
+        evaluations = ProductPageEvaluation.objects.with_total_creepiness()
+        my_evaluation = evaluations.get(pk=1)
+        my_evaluation.total_creepiness
+        ```
+
+        Returns:
+            int: Sum of creepiness value for all votes for this product evaluation.
+        """
         try:
             return self._total_creepiness
         except AttributeError:
-            return sum([vote.value for vote in self.votes.all()])
+            raise ValueError(
+                "Can't calculate total creepiness without `_total_creepiness` annotation. "
+                "Make sure to annotate the evaluation queryset by calling `.with_total_creepiness()` "
+                "method before accessing this property."
+            )
```

Comment:
I'm not sure if `ValueError` is the best type of exception here as it is more of a "Don't do this thing or we'll get in trouble" :thinking:

Reply:
I think perhaps re-raising `AttributeError` might make sense - you're accessing an attribute and it doesn't exist, just for a special reason that you're explaining well in the exception.

Perhaps also worth using exception chaining here to keep the trace from the original exception:

```python
...
except AttributeError as e:
   raise AttributeError("...") from e
```

Reply:
Done :+1: